web/src/features/datasets/lib/csvHelpers.ts

```diff
@@ -0,0 +1,214 @@
+import { parse } from "csv-parse";
+import { Readable } from "stream";
+import { type Prisma } from "@langfuse/shared";
+import { type StoredFile } from "@/src/features/datasets/server/tempStorage";
+
+export const MAX_PREVIEW_ROWS = 10;
+
+type ColumnType =
+  | "string"
+  | "number"
+  | "boolean"
+  | "null"
+  | "json"
+  | "array"
+  | "unknown"
+  | "mixed";
+
+export type CsvColumnPreview = {
+  name: string;
+  samples: string[];
+  inferredType: ColumnType;
+};
+
+export type CsvPreviewResult = {
+  fileName: string;
+  columns: CsvColumnPreview[];
+  totalColumns: number;
+  previewRows: number;
+  fileId?: string;
+};
+
+type ParseCsvOptions = {
+  preview?: number;
+  onHeader?: (headers: string[]) => void | Promise<void>;
+  onRow?: (
+    row: string[],
+    headers: string[],
+    rowIndex: number,
+  ) => void | Promise<void>;
+  collectSamples?: boolean;
+};
+
+const FORMULA_TRIGGERS = ["=", "+", "-", "@", "\t", "\r", "\n"];
+
+function sanitizeCsvValue(value: string): string {
+  // Convert to string and trim
+  const str = String(value).trim();
+
+  // Check for formula triggers at start
+  if (FORMULA_TRIGGERS.some((trigger) => str.startsWith(trigger))) {
+    // Prefix with single quote to prevent formula execution
+    return `'${str}`;
```

Comment:
where exactly could formula execution be triggered? As we insert via prisma, and prisma sends values as prepared statements, the insert should be safe in any way